workflow/rocoto/rocoto.py

```diff
@@ -109,12 +109,14 @@ def create_task(task_dict: Dict[str, Any]) -> List[str]:
             strings.append(f'\t{e}\n')
         strings.append('\n')
 
-    if dependency is not None:
+    if len(dependency) > 0:
```

Comment:
You are checking the length of `dependency`, but if it is `None` it has no length and will throw an exception. I tested in an interactive session to make sure this wasn't an instance of it being considered "truthy" enough to work:

```
>>> d=None
>>> if len(d) > 0:
...     print(True)
... else:
...     print(False)
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'NoneType' has no len()
```

The default value for `dependency` should be an empty list now instead of `None`. Also need to check all the calls to this function to make sure none of the tasks are passing `None` in.